src/corner-algorithms.jl

```diff
@@ -145,3 +145,43 @@ function gammacovs(cov_xx, cov_xy, cov_yy, gamma::Float64 = 1.4)
 
     filt_cov_xx, filt_cov_xy, filt_cov_yy
 end
+
+
+"""
+```
+moravec_response = moravec(img; [threshold], [window_size])
+```
+Compute the corner response values for each pixel in the input image using the Moravec corner detection algorithm.
+
+    # Arguments
+    - `img::AbstractArray`: The input image.
+    - `window_size::Int`: Size of the window for calculating the corner response.
+    
+    # Keyword Arguments
+    - `args...`: Additional arguments passed to the `gradcovs` function.
+    # Resources
+    - https://vincmazet.github.io/bip/detection/corners.html
+    - http://www0.cs.ucl.ac.uk/staff/g.brostow/classes/IP2008/L7_CornerDetection.pdf
+
+"""
+
+function moravec(img::AbstractArray; window_size::Int = 3, args...)
+    gradient_x, gradient_y = gradcovs(img, args...)
+    
+    corner_response = similar(img, Float64)  # Initialize a corner response image
+
+    for y in 1:size(img, 1) - window_size + 1
+        for x in 1:size(img, 2) - window_size + 1
+            min_sum_diff = Inf
+            for dy in -1:1
+                for dx in -1:1
+                    sum_diff = sum((gradient_x[y:y+window_size-1, x:x+window_size-1] .- gradient_x[y+dy:y+dy+window_size-1, x+dx:x+dx+window_size-1]).^2)
```

Comment:
There is an off by one error somewhere here. You are indexing a 3x3 matrix at 0:2 instead of 1:3 somehow.